pkg/client/results/junit.go

```diff
@@ -45,84 +124,119 @@ func Filter(predicate func(testCase reporters.JUnitTestCase) bool, testSuite rep
 	return out
 }
 
-// AlphabetizedTestCases implements Sort over the list of testCases.
-type AlphabetizedTestCases []reporters.JUnitTestCase
+// JUnitAlphabetizedTestCases implements Sort over the list of testCases.
+type JUnitAlphabetizedTestCases []JUnitTestCase
 
-func (a AlphabetizedTestCases) Len() int           { return len(a) }
-func (a AlphabetizedTestCases) Less(i, j int) bool { return a[i].Name < a[j].Name }
-func (a AlphabetizedTestCases) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a JUnitAlphabetizedTestCases) Len() int           { return len(a) }
+func (a JUnitAlphabetizedTestCases) Less(i, j int) bool { return a[i].Name < a[j].Name }
+func (a JUnitAlphabetizedTestCases) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 
 // predicate functions
 
-// Skipped returns true if the test was skipped.
-func Skipped(testCase reporters.JUnitTestCase) bool { return testCase.Skipped != nil }
+// JUnitSkipped returns true if the test was skipped.
+func JUnitSkipped(testCase JUnitTestCase) bool { return testCase.SkipMessage != nil }
 
-// Passed returns true if the test passed.
-func Passed(testCase reporters.JUnitTestCase) bool {
-	return testCase.Skipped == nil && testCase.FailureMessage == nil
+// JUnitPassed returns true if the test passed.
+func JUnitPassed(testCase JUnitTestCase) bool {
+	return testCase.SkipMessage == nil && testCase.Failure == nil
 }
 
-// Failed returns true if the test failed.
-func Failed(testCase reporters.JUnitTestCase) bool {
-	return testCase.Skipped == nil && testCase.FailureMessage != nil
+// JUnitFailed returns true if the test failed.
+func JUnitFailed(testCase JUnitTestCase) bool {
+	return testCase.SkipMessage == nil && testCase.Failure != nil
 }
 
-func processJunitFile(pluginDir, currentFile string) (Item, error) {
-	junitResults := reporters.JUnitTestSuite{}
+func junitProcessFile(pluginDir, currentFile string) (Item, error) {
 	relPath, err := filepath.Rel(pluginDir, currentFile)
 	if err != nil {
 		logrus.Errorf("Error making path %q relative to %q: %v", pluginDir, currentFile, err)
 		relPath = currentFile
 	}
 
-	// Passed unless a failure is encountered for aggregate status.
 	resultObj := Item{
 		Name:     filepath.Base(currentFile),
-		Status:   StatusPassed,
+		Status:   StatusUnknown,
 		Metadata: map[string]string{"file": relPath},
 	}
 
 	infile, err := os.Open(currentFile)
 	if err != nil {
-		resultObj.Status = StatusUnknown
 		resultObj.Metadata["error"] = err.Error()
+		resultObj.Status = StatusUnknown
+
 		return resultObj, errors.Wrapf(err, "opening file %v", currentFile)
 	}
 	defer infile.Close()
 
-	decoder := xml.NewDecoder(infile)
-	if err := decoder.Decode(&junitResults); err != nil {
-		resultObj.Status = StatusUnknown
-		resultObj.Metadata["error"] = err.Error()
-		return resultObj, errors.Wrap(err, "error decoding json into object")
+	resultObj, err = junitProcessReader(
```

Comment:
Removed all the file handling and put the rest of the logic into this new method.